### PR TITLE
Fixes #8764 - adds description to taxonomies

### PR DESCRIPTION
**Pull request by orrabin**

### CODE REVIEW

**Comment by dLobatog** [comment]
Looks good to me, is there any way to make the field look 'optional'?

**Comment by orrabin** [comment]
@eLobato it really isn't that clear that only name is required since there are only 3 fields when creating a location/organization.
Maybe @tbrisker will have a suggestion?

File: app/views/taxonomies/_form.html.erb [hunk 1]
```diff
@@ -10,6 +10,7 @@
                }) if show_parent?(taxonomy) %>
 
   <%= text_f f, :name %>
+  <%= text_f f, :description %>
```
Comment (ohadlevy) [inline app/views/taxonomies/_form.html.erb]: text area ?

File: app/views/taxonomies/index.html.erb [hunk 2]
```diff
@@ -14,6 +14,7 @@
 <table class="table table-bordered table-striped">
   <tr>
     <th><%= sort :title, :as => s_("Taxonomy|Name") %></th>
+    <th><%= sort :description, :as => s_("Taxonomy|Description") %></th>
```
Comment (ohadlevy) [inline app/views/taxonomies/index.html.erb]: not sure if we should show description in index?

**Comment by thomasmckay** [comment]
don't forget corresponding katello p-r

File: app/views/taxonomies/index.html.erb (hunk 2 above)
Comment (orrabin) [inline app/views/taxonomies/index.html.erb]: should I leave the scoped_search on description?

**Comment by orrabin** [comment]
@thomasmckay thanks, there are already changes that will become a PR later today

**Comment by tbrisker** [comment]
@orrabin @eLobato I think there isn't any good way to show a field as optional only for this form; If we decide on one we should apply it across the entire UI to be consistent. 
I think the name should be above the parent, as it's confusing to have the only required field (and therefore most important one) be second & preselected. (This also applies to hostgroups - maybe I should create a different issue for it?)

**Comment by dLobatog** [comment]
[test]

**Comment by dLobatog** [comment]
:+1: from my side. Waiting on https://github.com/Katello/katello/issues/4896 to merge.

File: app/views/taxonomies/_form.html.erb [hunk 3]
```diff
@@ -10,6 +10,7 @@
                }) if show_parent?(taxonomy) %>
 
   <%= text_f f, :name %>
+  <%= textarea_f f, :description, :rows => 5 %>
```
Comment (ohadlevy) [inline app/views/taxonomies/_form.html.erb]: can you please add all of this into a primary tab? it looks very strange when the other tabs are pushed down.

File: app/views/taxonomies/_form.html.erb [hunk 4]
```diff
@@ -1,19 +1,11 @@
...
-               { :label => _('Parent'), :onchange => 'parent_taxonomy_changed(this);',
-                 :help_inline => :indicator,
-                 :'data-url' => (controller_name == 'organizations' ? parent_taxonomy_selected_organization_path(taxonomy.id) :
-                                                                      parent_taxonomy_selected_location_path(taxonomy.id))
-               }) if show_parent?(taxonomy) %>
-
-  <%= text_f f, :name %>
   <div class="form-group row">
     <ul class="nav nav-pills nav-stacked col-md-3" data-tabs="pills">
+      <li class="active"><a href="#primary" data-toggle="tab"><%= _("Primary") %></a></li>
```
Comment (dLobatog) [inline app/views/taxonomies/_form.html.erb:6]: Maybe "Location" or "Organization" is a bit more clear than Primary?

Comment (orrabin) [inline app/views/taxonomies/_form.html.erb:6]: @eLobato is it worth checking the controller_name like in `data-url` to get the name right for location/organization? 
Is there a simpler way to get where I came from and give the correct name? 
Primary was picked so it's the same in both location and organization.

**Comment by daviddavis** [comment]
[test]

**Comment by dLobatog** [comment]
Merged as 581d3b6b36f5f4ec1251ae4ebbf4964843d2d3bb, thanks @orrabin and @daviddavis for reviewing/merging the Katello part.

**Pull request closed without merging**